airbyte-integrations/connectors/source-harvest/source_harvest/spec.json

```
@@ -25,9 +19,106 @@
         "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$",
         "examples": ["2017-01-25T00:00:00Z"],
         "type": "string"
+      },
+      "credentials": {
+        "title": "Authentication mechanism",
+        "description": "Choose how to authenticate to Harvest",
+        "type": "object",
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "Authenticate via Harvest (Oauth)",
+            "required": ["client_id", "client_secret", "refresh_token"],
+            "additionalProperties": false,
+            "properties": {
+              "auth_type": {
+                "type": "string",
+                "const": "Client",
+                "enum": ["Client"],
+                "default": "Client",
+                "order": 0
+              },
+              "client_id": {
+                "title": "Client ID",
+                "type": "string",
+                "description": "The Client ID of your application"
+              },
+              "client_secret": {
+                "title": "Client Secret",
+                "type": "string",
+                "description": "The client secret of your application",
+                "airbyte_secret": true
+              },
+              "refresh_token": {
+                "title": "Refresh Token",
+                "type": "string",
+                "description": "A refresh token generated using the above client ID and secret",
+                "airbyte_secret": true
+              }
+            }
+          },
+          {
+            "type": "object",
+            "title": "Authenticate with Personal Access Token",
+            "required": ["api_token"],
+            "additionalProperties": false,
+            "properties": {
+              "auth_type": {
+                "type": "string",
+                "const": "Token",
+                "enum": ["Token"],
+                "default": "Token",
+                "order": 0
+              },
+              "api_token": {
+                "title": "Personal Access Token",
+                "description": "Log into Harvest and then create new <a href=\"https://id.getharvest.com/developers\"> personal access token</a>.",
+                "type": "string",
+                "airbyte_secret": true
+              }
+            }
+          }
+        ]
       }
     }
   },
   "supportsIncremental": true,
-  "supported_destination_sync_modes": ["append"]
+  "supported_destination_sync_modes": ["append"],
+  "authSpecification": {
+    "auth_type": "oauth2.0",
+    "oauth2Specification": {
+      "rootObject": ["credentials", 0],
+      "oauthFlowInitParameters": [["client_id"], ["client_secret"]],
+      "oauthFlowOutputParameters": [["refresh_token"]]
+    }
+  },
+  "advancedAuth": {
+    "auth_flow_type": "oauth2.0",
+    "oauth_config_specification": {
+      "complete_oauth_output_specification": {
```

```suggestion
    "oauth_config_specification": {
      "predicate_key": ["credentials", "auth_type"],
      "predicate_value": "Client",      
      "complete_oauth_output_specification": {
```

---

is it mistake here ?
I see that `predicate_key` not inside `oauth_config_specification`

https://github.com/airbytehq/airbyte/blob/c5a726737891b5802eaf1515b8503f75eb56bbcb/airbyte-api/src/main/openapi/config.yaml#L3255

---

oh you're right then it's:

```suggestion
    "predicate_key": ["credentials", "auth_type"],
    "predicate_value": "Client",      
    "oauth_config_specification": {
      "complete_oauth_output_specification": {
```

---

fixed